client/src/actions/bulk.js

```diff
@@ -45,15 +44,33 @@ function buildFormData(form) {
     return formData;
 }
 
-function doAction(action, form) {
+function doAction(action, form, isAsyncJob) {
     return async dispatch => {
         dispatch(pending(action));
         try {
-            const result = await fetch('/api/monument/bulk/' + action, {
+            let result = await (await fetch('/api/monument/bulk/' + action + (isAsyncJob ? '/start' : ''), {
                 method: 'post',
                 body: buildFormData(form)
-            });
-            dispatch(success(action, await result.json()));
+            })).json();
+
+            if (isAsyncJob) {
+                const jobId = result.id;
+
+                let interval;
+                await new Promise(resolve => {
+                    interval = window.setInterval(async () => {
+                        result = await (await fetch(`/api/monument/bulk/${action}/progress/${jobId}`)).json();
+                        dispatch(pending(action, result.progress));
+
+                        if (result.future.done) resolve();
+                    }, 200);
+                });
+                window.clearInterval(interval);
+
+                result = await (await fetch(`/api/monument/bulk/${action}/result/${jobId}`)).json();
+            }
```

Comment:
This is where we start the job and get progress of it (every 200ms currently), then once the job is done get the final results and send them to the reducer